### ISSUE TITLE
History Navigation work around

### ISSUE THREAD

**Issue by TanviHacks** [mozilla/contain-facebook#24]
Step 1 - user browses internet in Tab 1.
Step 2 - user navigates to facebook.com.  browser opens facebook container Tab 2 and closes Tab 1.
Step 3 - user navigates to non-facebook.com site and browser opens regular Tab 3 and closes Tab 2.

In Step 2, we don't have to close Tab 1.  The issue is to stop closing Tab 1 in Step 2.

Here are more details:
In Step 3, we close Tab 2 because we don't want facebook.com open and lurking around in a background tab unnecessarily.  The addons purpose is privacy protection against facebook, so letting it keep running doesn't help with that.

But there is no reason to delete the pre-facebook history.

This isn't a big UX issue, since the user knows that the addon is isolating Facebook into its own tab.  So if they type "facebook.com" in the url bar and see a new tab open, the user knows that this is the addon doing its job of isolating facebook.

**Comment by hackel** [comment]
This is a rather frustrating issue.  To suddenly have your entire tab's history erased because you happened to click on a link to facebook.com is unexpected and inconvenient.  Whenever this has come up, it's been searching for a name, a person or business, so I click the link to Facebook, discover it's not the one I want, then go to hit back to resume my search and can't.  I have to repeat the search or dig through my history to find it.

Until Firefox supports traversing containers with the back/forward buttons (the ideal solution), closing an existing tab should be considered data loss and must never happen!

**Comment by forteller** [comment]
I totally agree with hackel, and came to Github to file a bug about it. At least there should be an option to select if clicking a Facebook link should open it in a new tab in addiotion to the tab you're in when you click it, so that you can keep your history.

Thanks for your important work!

**Comment by pmorch** [comment]
The title of this issue contains "work around". But there is no work around.  I suggest renaming the issue: "History Navigation Broken".

If I search for something and have facebook among the search results, then visiting the facebook search result means I can't go back to the search results? This breaks the UX for me at least.

**Comment by petteyg** [comment]
This. I'm using Google to search, and a lot of the results I want to look at are on Facebook's Help pages. Every time I click one of those results, I have to right-click and "undo close tab" because I can't click "back" in the new container tab.

**Comment by shbach** [comment]
I also came here to file this bug. It's a really annoying issue. Just don't close the previous tab or at least give the option to turn off this behavior.

**Comment by ygoe** [comment]
The duplicate issue contains a suggested fix to this annoyance. https://github.com/mozilla/contain-facebook/issues/143#issuecomment-627241868 Also it has been argued that the issue isn't minor but could be considered loss of state, if not loss of data. And the issue subject is wrong.

Is somebody still working on this add-on?

**Comment by maxxcrawford** [comment]
@ygoe We don't have any targeted releases scheduled right now. Please feel free to open a PR!

**Comment by martsyn** [comment]
Is this issue going to be fixed? I feel if FF team shoves this addon down everyone's throat as a recommendation, this bug should be fixed.

How about just leaving Tab 1 open if passing history cannot be done between tabs?

**Comment by maxxcrawford** [comment]
@martsyn This project is currently in maintenance, so progress from our team will be slow. However, I've been working on a [settings panel PR](https://github.com/mozilla/contain-facebook/pull/721) that accounts for this behavior. 

Specifically, matching the same UX as Multi-account Containers options as outlined here: https://github.com/mozilla/multi-account-containers/pull/1835

**Comment by macks2008** [comment]
Current workaround: the tab/history being replaced is sent to `History > Recently Closed Tabs` and not completely discarded. However, the UX here is still not great and I would like to know if any progress has been made on resolving this fully.

**Comment by bfsh00t** [comment]
5 years and a half later, this most annoying behavior still exists. The fact that this extension is recommended by mozilla and still hasn't been solved after more than 5 years makes you wish it was a priority.
Sweeping the dust under the carpet is not a solution worthy of a top-class web browser 🥴
